src/ray/raylet/scheduling_queue.cc

```diff
@@ -111,34 +107,48 @@ bool SchedulingQueue::TaskQueue::RemoveTask(const TaskID &task_id,
   return true;
 }
 
-bool SchedulingQueue::TaskQueue::HasTask(const TaskID &task_id) const {
+bool TaskQueue::HasTask(const TaskID &task_id) const {
   return task_map_.find(task_id) != task_map_.end();
 }
 
-const std::list<Task> &SchedulingQueue::TaskQueue::GetTasks() const { return task_list_; }
+const std::list<Task> &TaskQueue::GetTasks() const { return task_list_; }
 
-const ResourceSet &SchedulingQueue::TaskQueue::GetCurrentResourceLoad() const {
+const ResourceSet &TaskQueue::GetCurrentResourceLoad() const {
   return current_resource_load_;
 }
 
+bool ReadyQueue::AppendTask(const TaskID &task_id, const Task &task) {
+  auto &resources = task.GetTaskSpecification().GetRequiredResources();
+  RAY_CHECK(tasks_with_resources_[resources].insert(task_id).second == true);
+  return TaskQueue::AppendTask(task_id, task);
+}
+
+bool ReadyQueue::RemoveTask(const TaskID &task_id, std::vector<Task> *removed_tasks) {
+  if (task_map_.find(task_id) != task_map_.end()) {
+    auto &resources = task_map_[task_id]->GetTaskSpecification().GetRequiredResources();
+    RAY_CHECK(tasks_with_resources_[resources].erase(task_id) == 1);
```

Comment:
same, this will crash the raylet if there's an inconsistency found between the `task_map_` and `tasks_with_resources_`. While this inconsistency shouldn't happen, I'd rather we clean up the inconsistency, log a warning, return false, but not make it fatal.

Reply:
Same here, the check should absolutely be fatal since if it fails, there is a bug in the code.